### ISSUE TITLE
Tries to open but immediately closes (3.0)

### ISSUE THREAD

**Issue by mlfarrell** [jbtule/cdto#40]
Only works in some folders.  Most of the time immediately closes and doesn't launch

**Comment by mlfarrell** [comment]
Looks like this fixed it.  At least mostly

```int main(int argc, const char * argv[]) {
    @autoreleasepool {
        // Setup code that might create autoreleased objects goes here.
        FinderApplication* finder = [SBApplication applicationWithBundleIdentifier:@"com.apple.Finder"];
        TerminalApplication* terminal = [SBApplication applicationWithBundleIdentifier:@"com.apple.Terminal"];
                        
        FinderItem *target = [(NSArray*)[[finder selection] get] firstObject];
        if (target == nil){
            target = [[[[finder FinderWindows] objectAtLocation:@0] target] get];
        }
        
        if ([[target kind] isEqualToString:@"Alias"]){
            target = (FinderItem*)[(FinderAliasFile*)target originalItem];
        }
        NSString* fileUrl = [target URL];
        
        if(fileUrl != nil && ![fileUrl hasSuffix:@"/"]){
            fileUrl = [fileUrl stringByDeletingLastPathComponent];
        }
        
        NSURL* url = [NSURL URLWithString:fileUrl];
                
        if (url != nil){
            [terminal activate];
            [terminal open:@[url]];
        } else {
            SBElementArray *finderWindows = finder.FinderWindows;
            FinderWindow *finderWindow = finderWindows.firstObject;
            FinderFolder *finderFolder = finderWindow.properties[@"target"];
            
            fileUrl = finderFolder.URL;
            url = [NSURL URLWithString:fileUrl];
            
            if(fileUrl)
            {
                [terminal activate];
                [terminal open:@[url]];
            }
        }
    }
}
```

**Comment by jbtule** [comment]
I think the problem was
```objective-c
            target = [[[[finder FinderWindows] objectAtLocation:@0] target] get];
```
As apple script is not zero based!! whoops!!

Since it looks like you've already tried the source, can you try pull request #42 see if it fixes it.

Thank you!

**Comment by jbtule** [comment]
Released 3.1, let me know if this fixes your issue. Thanks.

https://github.com/jbtule/cdto/releases/tag/v3.1

**Comment by YvesR** [comment]
@jbtule I had the same issue, thought it is catalina, but 3.1 fixed the issue. Works now again.
